rust/src/base.rs

```diff
@@ -484,6 +481,64 @@ impl<'a> Iterator for BeneficiariesIter<'a> {
     }
 }
 
+/// An endpoint to a consignment exchange medium.
+#[derive(
+    Clone,
+    Ord,
+    PartialOrd,
+    Eq,
+    PartialEq,
+    Hash,
+    Debug,
+    Display,
+    StrictEncode,
+    StrictDecode,
+)]
+#[cfg_attr(
+    feature = "serde",
+    derive(Serialize, Deserialize),
+    serde(crate = "serde_crate")
+)]
+#[display(inner)]
+#[non_exhaustive]
+pub enum ConsignmentEndpoint {
+    /// Storm protocol
+    #[display("storm:{0}")]
+    Storm(NodeAddr),
+
+    /// RGB HTTP JSON-RPC protocol
+    #[display("rgbhttpjsonrpc:{0}")]
+    RgbHttpJsonRpc(String), // Url,
```

Comment:
Nit: I think just `RgbJsonRpc` would work, `Http` part is obvious for `JsonRpc` protocol.